Source/player.cpp

```diff
@@ -3665,149 +3665,85 @@ void __fastcall MakePlrPath(int pnum, int xx, int yy, BOOL endspace)
-			switch ( v5 )
-			{
-				case PC_WARRIOR:
-					v4 = PS_WARR27;
-					goto LABEL_53;
-				case PC_ROGUE:
-					v4 = PS_ROGUE27;
-					goto LABEL_53;
-				case PC_SORCERER:
-					v4 = PS_MAGE27;
-					goto LABEL_53;
-			}
-			return;
-		}
-		if ( pcurs != CURSOR_HAND
-		  || MouseY >= 352
-		  || (chrflag && MouseX < 320 || invflag && MouseX > 320)
-		  && v2 != 2
-		  && v2 != 5
-		  && v2 != 26
-		  && v2 != 9
-		  && v2 != 27 )
-		{
-			return;
-		}
-		_LOBYTE(v1) = *((_BYTE *)&plr[0]._pRSplType + v1);
-		if ( (v1 & 0x80u) != 0 )
-			goto LABEL_46;
-		if ( (char)v1 <= 1 )
-		{
-			v6 = CheckSpell(v0, v2, v1, 0);
+
+	int rspell = plr[myplr]._pRSpell;
+	if ( rspell == SPL_INVALID ) {
+		if ( plr[myplr]._pClass == PC_WARRIOR ) {
+			PlaySFX(PS_WARR34);
+		} else if ( plr[myplr]._pClass == PC_ROGUE ) {
+			PlaySFX(PS_ROGUE34);
+		} else if ( plr[myplr]._pClass == PC_SORCERER ) {
+			PlaySFX(PS_MAGE34);
 		}
-		else
-		{
-			if ( (_BYTE)v1 != 2 )
-			{
-				if ( (_BYTE)v1 == 3 )
-				{
-					v6 = UseStaff();
-					goto LABEL_36;
-				}
-LABEL_46:
-				if ( _LOBYTE(plr[v0]._pRSplType) == 1 )
-				{
-					v15 = plr[v0]._pClass;
-					switch ( v15 )
-					{
-						case PC_WARRIOR:
-							v4 = PS_WARR35;
-							goto LABEL_53;
-						case PC_ROGUE:
-							v4 = PS_ROGUE35;
-							goto LABEL_53;
-						case PC_SORCERER:
-							v4 = PS_MAGE35;
-							goto LABEL_53;
-					}
-				}
-				return;
-			}
-			v6 = UseScroll();
+		return;
+	}
+
+	if ( leveltype == DTYPE_TOWN && !spelldata[rspell].sTownSpell ) {
+		if ( plr[myplr]._pClass == PC_WARRIOR ) {
+			PlaySFX(PS_WARR27);
+		} else if ( plr[myplr]._pClass == PC_ROGUE ) {
+			PlaySFX(PS_ROGUE27);
+		} else if ( plr[myplr]._pClass == PC_SORCERER ) {
+			PlaySFX(PS_MAGE27);
 		}
-LABEL_36:
-		v0 = myplr;
-		if ( v6 )
-		{
-			v7 = plr[myplr]._pRSpell;
-			if ( v7 == 6 )
-			{
-				v8 = GetDirection(plr[myplr].WorldX, plr[myplr].WorldY, cursmx, cursmy);
-				v9 = GetSpellLevel(myplr, plr[myplr]._pRSpell);
-				v10 = 21720 * myplr;
-				_LOWORD(v10) = plr[myplr]._pRSpell;
-				NetSendCmdLocParam3(1u, CMD_SPELLXYD, cursmx, cursmy, v10, v8, v9);
-			}
-			else if ( pcursmonst == -1 )
-			{
-				if ( pcursplr == -1 )
-				{
-					v13 = GetSpellLevel(myplr, v7);
-					v14 = 21720 * myplr;
-					_LOWORD(v14) = plr[myplr]._pRSpell;
-					NetSendCmdLocParam2(1u, CMD_SPELLXY, cursmx, cursmy, v14, v13);
-				}
-				else
-				{
-					v12 = GetSpellLevel(myplr, v7);
-					NetSendCmdParam3(1u, CMD_SPELLPID, pcursplr, plr[myplr]._pRSpell, v12);
-				}
-			}
-			else
-			{
-				v11 = GetSpellLevel(myplr, v7);
-				NetSendCmdParam3(1u, CMD_SPELLID, pcursmonst, plr[myplr]._pRSpell, v11);
-			}
-			return;
+		return;
+	}
+
+	if ( pcurs != CURSOR_HAND
+	  || MouseY >= 352
+	  || (chrflag && MouseX < 320 || invflag && MouseX > 320)
+	  && rspell != SPL_HEAL
+	  && rspell != SPL_IDENTIFY
+	  && rspell != SPL_REPAIR
+	  && rspell != SPL_INFRA
+	  && rspell != SPL_RECHARGE
+	) {
+		return;
+	}
+
+	UCHAR addflag = FALSE;

```

Comment:
Looks like this should be BOOL

Reply:
With BOOL (32bit) it ends up as different ASM, i might could make it bool as that is 8bit and should still be exact.

Reply:
Thanks I totally missed that the 2td and 3td where bool and not BOOL. I also found another error where sTownSpell was also not set to BOOL but unsigned char, both are fixed now.